### PR TITLE
Update to Gnome45

### PR DESCRIPTION
**Pull request by gilvbp**
Update to Gnome45.
TO-DO on prefs.js needs to update the widget with the new adw.PreferencesGroup layout attributes

### CODE REVIEW

**Comment by GrylledCheez** [comment]
@gilvbp, do we need keybindings.js at all?

And eslint is complaining about your use of import and export. But that might just be that we need a newer eslintrc.yml

File: extension.js [hunk 1]
```diff
@@ -68,12 +66,12 @@ function updateSettings() {
 }
 
 const wintile = {
-    extdatadir: imports.misc.extensionUtils.getCurrentExtension().path,
-    shell_version: parseInt(Config.PACKAGE_VERSION.split('.')[1], 10),
+    extdatadir: Extension.path,
+//    shell_version: parseInt(Config.PACKAGE_VERSION.split('.')[1], 10),
```
[the remark this answers is not in the document]
Comment (gilvbp) [reply extension.js:70]: It's only for 45.

**Comment by gilvbp** [comment]
> @gilvbp, do we need keybindings.js at all?
> 
> And eslint is complaining about your use of import and export. But that might just be that we need a newer eslintrc.yml

Yes, we need. We have to change the eslintrc.yml to accept new ecma import type.

**Comment by GrylledCheez** [comment]
@gilvbp, use the latest .eslingrc.yml I pushed to your repo, then run eslint with --fix as described in CONTRIBUTING.md

**Comment by GrylledCheez** [comment]
That fixes the style issues!

I don't have anything with Gnome 45 right now. So just do the following checks and I'll sign off on the PR.

Perform these tests in both landscape and portrait
- [ ] keyboard moving
- [ ] mouse moving around edges
- [ ] ctrl+click and drag/drop
- [ ] click drag, then ctrl+super and drag/drop
- [ ] crtl+super arrow to grow in that direction 
- [ ] 1-5 rows
- [ ] 1-5 columns

**Comment by GrylledCheez** [comment]
If you want, I'll accept this partial PR while you try and get this working. But we aren't going to release until the prefs.js works, 

But here's the preferences guide for 45.
https://gjs.guide/extensions/development/preferences.html#gsettings

**Comment by gilvbp** [comment]
> If you want, I'll accept this partial PR while you try and get this working. But we aren't going to release until the prefs.js works,
> 
> But here's the preferences guide for 45. https://gjs.guide/extensions/development/preferences.html#gsettings
Please accept this partial PR.
I have in mind that I need to get pref.js working to release it.
Thanks for the link, but I already read G45 preferences. The problem is that the widget has changed a lot of layout attributes, as you can see here (https://docs.gtk.org/gtk4/class.Widget.html) and here: https://gnome.pages.gitlab.gnome.org/libadwaita/doc/main/class.PreferencesGroup.html)
So, we will have a lot of refactoring.

**Comment by GrylledCheez** [comment]
Great work. I'll see what I can learn this weekend. Make sure you update the README to say that this is only for 45, and to give yourself credit.

**Comment by atassis** [comment]
Hi there. I am a typescript developer and can help with this PR if needed. Migrated to latest fedora with gnome 45 and without this extension it doesnt feel good to use it. So I can both test or do smth else. Tried to write a new PR by myself but it is not that comfortable to debug what is wrong with extension changes

**Comment by GrylledCheez** [comment]
@atassis, take a look at some of the groundwork I started in the develop branch instead of this PR.

**Pull request closed without merging**